design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,129 @@
+# Rook-Ceph Object Bucket Controller
+
+## Proposed Feature
+
+Currently, the automation exists in the Rook-Ceph operator to support [Ceph Object Store](../object-store.md) and [Ceph Object Store User](https://github.com/rook/rook/issues/1583) custom resource definitions (CRDs).  
+
+This design closes the loop for Rook-Ceph object stores with a CephObjectBucket controller and CRD.  Users will request buckets via the Kubernetes API and have returned ConfigMaps with connection information for that bucket.  One ConfigMap will be created per CephObjectBucket.  ConfigMap names will be deterministic and based on the name of the CephObjectBucket.  This gives users the opportunity to define their Pod spec at the same time as the CephObjectBucket.  Additionally, the CephObjectBucket and Pod can safely be deployed at the same time thanks to built in synchronization in Kubernetes.
+
+Bucket deletion in the early stages of this controller design will not be addressed beyond cleaning up generated resources.  Deleting a CephObjectBucket will only cause deletion of the associated ConfigMap.  The controller will not delete the bucket from the object store. 
```

Comment:
That's correct.  The consensus in early drafts was that it was preferable to suspend the bucket on API resource deletion.  Ceph seems to supply this functionality.  The reason being that it would be far too easy to accidentally destroy your buckets.